src/event_loop.rs

```diff
@@ -426,6 +427,65 @@ impl<T: 'static> fmt::Debug for EventLoopProxy<T> {
     }
 }
 
+/// Errors that may occur when trying to create the Winit event loop.
+#[derive(Debug)]
+#[non_exhaustive]
+pub enum EventLoopCreationError {
+    /// The event loop can't be re-created.
+    RecreationAttempt,
+    /// The OS cannot perform the operation.
+    Os(OsError),
+}
+
+impl fmt::Display for EventLoopCreationError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
+        match self {
+            Self::RecreationAttempt => write!(f, "tried to recreate EventLoop"),
+            Self::Os(e) => write!(f, "{e}"),
```

Comment:
Generally, it compiles faster if you instead write:

```rust
fmt::Display::fmt(e, f)
```

Goes for the rest of these kinds of errors as well.

Reply:
Compilation speed in the minor like this is not really a concern IMO, and I prefer the consistency (e.g. you write it very similar as for `Self::RecreationAttempt` and `Self::Os`).

But if you insist, then sure, I don't really care either way.

Reply:
I rewrote this one to give a more specific error message instead